manim/mobject/graphing/number_line.py

```diff
@@ -279,7 +279,7 @@ def add_ticks(self):
         elongated_tick_size = self.tick_size * self.longer_tick_multiple
         for x in self.get_tick_range():
             size = self.tick_size
-            if x in self.numbers_with_elongated_ticks:
+            if np.any(np.isclose(x, self.numbers_with_elongated_ticks)):
```

Comment:
I am somewhat worried about the default precision of `np.isclose`, what if someone wants to work with a number line in the 10^-5 range? Is there a better way that respects the current number line scale?

Reply:
Wouldn't a numberline in the 10^-5 range be basically invisible? In any case, the 10^-5 default is the relative tolerance, so it respects the scale of the numberline already. We can perhaps make the absolute tolerance smaller if you wish, although I don't think it matters.

Reply:
This illustrates what I mean perhaps a bit better: currently, the scene

```py
def construct(self):
    nl = NumberLine(
        x_range=[1e-5, 2e-5, 1e-6],
        length=6,
        numbers_with_elongated_ticks=[12e-6, 17e-6],
        include_numbers=True,
        include_tip=True,
    )
    self.add(nl)
```

renders as 

![image](https://github.com/ManimCommunity/manim/assets/11851593/d22b8576-15d8-47dc-8158-3c3142424e64)

The NumberLine is very well visible (although the tick labels are useless in this particular form, they'd need to be formatted). With your solution, all ticks would be elongated (in contrast to none, which is also wrong and the case here). While your solution is better than we have currently, it still has the problem of not properly respecting the scale.

Reply:
Interesting - so the length is the scaling factor. That makes sense. As I said before, the `10^-5` is the relative tolerance (`np.isclose` uses the equation `abs(a - b) <= (absolute_tolerance + relative_tolerance * abs(b))`), and `10^-8` is the absolute tolerance to ensure correctness very close to `0`. You can see this if you render the scene you gave with my changes, which correctly gives the 3rd and 8th ticks elongated:

![image](https://github.com/ManimCommunity/manim/assets/20744112/be600156-8964-4ed4-9e34-a26fef7488fc)

Reply:
True, sorry -- I oversimplified the example that I originally tested with, which used
```
        x_range=[1 + 1e-5, 1 + 2e-5, 1e-6],
        numbers_with_elongated_ticks=[1 + 12e-6, 1 + 17e-6],
```
where your fix has the behavior I've described.

Reply:
Ah yes - I see the issue, thanks. This should be fairly simple to fix - just subtracting the x_min should make this behave correctly.

Reply:
Thank you! Would you mind adding a (non-graphical) test to `tests/module/graphing/test_ticks.py`? 

It should suffice to check that for a created `NumberLine` from the pathological example above has `self.ticks` contains mobjects that satisfy something along the lines of `all(tick.height > self.tick_size if ind in [3, 8] else tick.height == self.tick_size for ind, tick in enumerate(self.ticks))` (please feel free to make this more readable or write this in a completely different way; this is just to give you an idea how this should be checked without rendering an image).

Reply:
Cool, added. One thing to note is there's a difference between the tick heights and the tick sizes; specifically the heights are twice the sizes due to `result = Line(size * DOWN, size * UP)` at `manim/mobject/graphing/number_line.py:306`. I've written the test in a way that avoids that issue for now.